chalice/deploy/newdeployer.py

```diff
@@ -214,17 +214,76 @@ def build(self, config, stage_name):
         resources = []  # type: List[models.Model]
         deployment = models.DeploymentPackage(models.Placeholder.BUILD_STAGE)
         for function in config.chalice_app.pure_lambda_functions:
-            new_config = config.scope(chalice_stage=config.chalice_stage,
-                                      function_name=function.name)
-            role = self._get_role_reference(new_config, stage_name, function)
-            resource = self._build_lambda_function(
-                new_config, function, deployment, role)
+            resource = self._create_lambda_model(config, deployment,
+                                                 function.name,
+                                                 function.handler_string,
+                                                 stage_name)
             resources.append(resource)
+        for event_source in config.chalice_app.event_sources:
+            scheduled_event = self._create_event_model(
+                config, deployment, event_source, stage_name)
+            resources.append(scheduled_event)
         return models.Application(stage_name, resources)
 
-    def _get_role_reference(self, config, stage_name, function):
-        # type: (Config, str, app.LambdaFunction) -> models.IAMRole
-        role = self._create_role_reference(config, stage_name, function)
+    def _create_event_model(self,
+                            config,        # type: Config
+                            deployment,    # type: models.DeploymentPackage
+                            event_source,  # type: app.CloudWatchEventSource
+                            stage_name,    # type: str
+                            ):
+        # type: (...) -> models.ScheduledEvent
+        lambda_function = self._create_lambda_model(
+            config, deployment, event_source.name,
+            event_source.handler_string, stage_name
+        )
+        # Resource names must be unique across a chalice app.
+        # However, in the original deployer code, the cloudwatch
+        # event + lambda function was considered a single resource.
+        # Now that they're treated as two separate resources we need
+        # a unique name for the event_source that's not the lambda
+        # function resource name.  We handle this by just appending
+        # '-event' to the name.  Ideally this is handled in app.py
```

Comment:
Because we've made the requirement that resource names are unique, some of the internal code is implemented to take advantage of that fact.  For example, you look up a deployed resource by its resource name.

I think we'll need to add validation that resource names are unique so a user gets a more clear error message.  We'll need this anyways.

My comment about moving this to the `app.py` would be that if a user doesn't provide a name, the *default* would be the `function_name + '-event'`.  However, if they explicitly provide a name then we leave it untouched.

Reply:
Talked about it offline. Let's just look into adding the validation in a subsequent PR.